skxray/fitting/base/parameter_data.py

```diff
@@ -130,6 +127,91 @@
+                  'e_offset': 'fixed',
+                  'e_quadratic': 'fixed',
+                  'fwhm_fanoprime': 'fixed',
+                  'fwhm_offset': 'fixed',
+                  'non_fitting_values': 'fixed'}
+
+e_calibration = {'coherent_sct_amplitude': 'none',
+                  'coherent_sct_energy': 'fixed',
+                  'compton_amplitude': 'none',
+                  'compton_angle': 'fixed',
+                  'compton_f_step': 'fixed',
+                  'compton_f_tail': 'fixed',
+                  'compton_fwhm_corr': 'fixed',
+                  'compton_gamma': 'fixed',
+                  'compton_hi_f_tail': 'fixed',
+                  'compton_hi_gamma': 'fixed',
+                  'e_linear': 'lohi',
+                  'e_offset': 'lohi',
+                  'e_quadratic': 'fixed',
+                  'fwhm_fanoprime': 'fixed',
+                  'fwhm_offset': 'fixed',
+                  'non_fitting_values': 'fixed'}
+
+linear = {'coherent_sct_amplitude': 'none',
+          'coherent_sct_energy': 'fixed',
+          'compton_amplitude': 'none',
+          'compton_angle': 'fixed',
+          'compton_f_step': 'fixed',
+          'compton_f_tail': 'fixed',
+          'compton_fwhm_corr': 'fixed',
+          'compton_gamma': 'fixed',
+          'compton_hi_f_tail': 'fixed',
+          'compton_hi_gamma': 'fixed',
+          'e_linear': 'fixed',
+          'e_offset': 'fixed',
+          'e_quadratic': 'fixed',
+          'fwhm_fanoprime': 'fixed',
+          'fwhm_offset': 'fixed',
+          'non_fitting_values': 'fixed'}
+
+free_more = {'coherent_sct_amplitude': 'none',
+               'coherent_sct_energy': 'lohi',
+               'compton_amplitude': 'none',
+               'compton_angle': 'lohi',
+               'compton_f_step': 'lohi',
+               'compton_f_tail': 'fixed',
+               'compton_fwhm_corr': 'lohi',
+               'compton_gamma': 'lohi',
+               'compton_hi_f_tail': 'fixed',
+               'compton_hi_gamma': 'fixed',
+               'e_linear': 'lohi',
+               'e_offset': 'lohi',
+               'e_quadratic': 'lohi',
+               'fwhm_fanoprime': 'lohi',
+               'fwhm_offset': 'lohi',
+               'non_fitting_values': 'fixed'}
+
+fit_with_tail = {'coherent_sct_amplitude': 'none',
+                 'coherent_sct_energy': 'lohi',
+                 'compton_amplitude': 'none',
+                 'compton_angle': 'lohi',
+                 'compton_f_step': 'fixed',
+                 'compton_f_tail': 'lohi',
+                 'compton_fwhm_corr': 'lohi',
+                 'compton_gamma': 'fixed',
+                 'compton_hi_f_tail': 'fixed',
+                 'compton_hi_gamma': 'fixed',
+                 'e_linear': 'lohi',
+                 'e_offset': 'lohi',
+                 'e_quadratic': 'lohi',
+                 'fwhm_fanoprime': 'lohi',
+                 'fwhm_offset': 'lohi',
+                 'non_fitting_values': 'fixed'}
+
```

Comment:
Look at how the update function now works. This is not an issue.

On Thu, Mar 5, 2015, 15:56 Li Li <notifications@github.com> wrote:

> In skxray/fitting/base/parameter_data.py
> <https://github.com/licode/scikit-xray/pull/5#discussion_r25905698>:
>
> > +                 'coherent_sct_energy': 'lohi',
> > +                 'compton_amplitude': 'none',
> > +                 'compton_angle': 'lohi',
> > +                 'compton_f_step': 'fixed',
> > +                 'compton_f_tail': 'lohi',
> > +                 'compton_fwhm_corr': 'lohi',
> > +                 'compton_gamma': 'fixed',
> > +                 'compton_hi_f_tail': 'fixed',
> > +                 'compton_hi_gamma': 'fixed',
> > +                 'e_linear': 'lohi',
> > +                 'e_offset': 'lohi',
> > +                 'e_quadratic': 'lohi',
> > +                 'fwhm_fanoprime': 'lohi',
> > +                 'fwhm_offset': 'lohi',
> > +                 'non_fitting_values': 'fixed'}
> > +
>
> @tacaswell <https://github.com/tacaswell> , I like the idea of keeping
> strategy as separate dict. However, when we do the fitting, the
> default_param dict will keep on growing as items related to elements are
> added. In that case, we also need to keep on growing each individual
> strategy dict to make them match same size of the default_param. I think
> this may not be easy to handle.
>
> Another issue is that those strategies are well tested, so the chance to
> touch them is quite low. So keeping the previous param_dict and maintaining
> only one dict is actually easy.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/licode/scikit-xray/pull/5/files#r25905698>.
>